{
	"id": "122846",
	"key": "ALOY-884",
	"fields": {
		"issuetype": {
			"id": "1",
			"description": "A problem which impairs or prevents the functions of the product.",
			"name": "Bug",
			"subtask": false
		},
		"project": {
			"id": "11113",
			"key": "ALOY",
			"name": "Alloy",
			"projectCategory": {
				"id": "10400",
				"description": "Tools for developing applications",
				"name": "Tooling"
			}
		},
		"fixVersions": [
			{
				"id": "15403",
				"description": "Alloy 1.3.0",
				"name": "Alloy 1.3.0",
				"archived": false,
				"released": true,
				"releaseDate": "2013-12-20"
			},
			{
				"id": "15755",
				"description": "2013 Sprint 24",
				"name": "2013 Sprint 24",
				"archived": true,
				"released": true,
				"releaseDate": "2013-11-29"
			}
		],
		"resolution": {
			"id": "1",
			"description": "A fix for this issue is checked into the tree and tested.",
			"name": "Fixed"
		},
		"resolutiondate": "2013-11-24T15:39:35.000+0000",
		"created": "2013-11-23T12:42:42.000+0000",
		"priority": {
			"name": "Critical",
			"id": "1"
		},
		"labels": [
			"alloy",
			"cli",
			"compile",
			"qe-port",
			"sync",
			"widgets"
		],
		"versions": [],
		"issuelinks": [
			{
				"id": "33344",
				"type": {
					"id": "10000",
					"name": "Blocks",
					"inward": "is blocked by",
					"outward": "blocks"
				},
				"inwardIssue": {
					"id": "122857",
					"key": "TIMOB-15806",
					"fields": {
						"summary": "Alloy compiler runs twice when installed via titanium-bundle",
						"status": {
							"description": "The issue is considered finished, the resolution is correct. Issues which are closed can be reopened.",
							"name": "Closed",
							"id": "6",
							"statusCategory": {
								"id": 3,
								"key": "done",
								"colorName": "green",
								"name": "Done"
							}
						},
						"priority": {
							"name": "Critical",
							"id": "1"
						},
						"issuetype": {
							"id": "1",
							"description": "A problem which impairs or prevents the functions of the product.",
							"name": "Bug",
							"subtask": false
						}
					}
				}
			}
		],
		"assignee": {
			"name": "tlukasavage",
			"key": "tlukasavage",
			"displayName": "Tony Lukasavage",
			"active": true,
			"timeZone": "America/Los_Angeles"
		},
		"updated": "2013-12-05T20:27:10.000+0000",
		"status": {
			"description": "The issue is considered finished, the resolution is correct. Issues which are closed can be reopened.",
			"name": "Closed",
			"id": "6",
			"statusCategory": {
				"id": 3,
				"key": "done",
				"colorName": "green",
				"name": "Done"
			}
		},
		"components": [
			{
				"id": "13605",
				"name": "Models"
			},
			{
				"id": "13604",
				"name": "Tooling",
				"description": "Items related to Alloy tooling and workflow"
			}
		],
		"description": "After upgrading to Alloy 1.3.0-alpha3 I get errors when building a project that worked fine with Alloy 1.2.x:\r\n\r\n{code}\r\n[DEBUG] ----- CLEANING RESOURCES -----\r\n[DEBUG] Removing orphaned controllers...\r\n[DEBUG] Removing orphaned models...\r\n[DEBUG] Removing orphaned styles...\r\n[DEBUG] Removing orphaned sync adapters...\r\n[DEBUG] fs.js:760\r\n[DEBUG]   return binding.unlink(pathModule._makeLong(path));\r\n[DEBUG]                  ^\r\n[ERROR] : EPERM, operation not permitted '/Users/zandbergen/dev/apps/VBG/Resources/iphone/alloy/sync/nl.fokkezb.titabs.gcal\r\n{code}\r\n\r\nThe problem seems to be that I have widgets that ship with their own custom sync adapter. The sync adapter is provided via:\r\n\r\n{code}\r\n/app/widgets/nl.fokkezb.titabs.gcal/assets/alloy/sync/restapi.js\r\n{code}\r\n\r\nAfter the build stops on this error, the file *does* exist under the displayed Resources path.",
		"attachment": [],
		"flagged": false,
		"summary": "Errors removing orphaned sync adapters when using a widget-specific sync adapter",
		"creator": {
			"name": "fokkezb",
			"key": "fokke",
			"displayName": "Fokke Zandbergen",
			"active": true,
			"timeZone": "Europe/Amsterdam"
		},
		"subtasks": [],
		"reporter": {
			"name": "fokkezb",
			"key": "fokke",
			"displayName": "Fokke Zandbergen",
			"active": true,
			"timeZone": "Europe/Amsterdam"
		},
		"environment": "Alloy 1.3.0-alpha3",
		"comment": {
			"comments": [
				{
					"id": "280871",
					"author": {
						"name": "tlukasavage",
						"key": "tlukasavage",
						"displayName": "Tony Lukasavage",
						"active": true,
						"timeZone": "America/Los_Angeles"
					},
					"body": "PR: https://github.com/appcelerator/alloy/pull/278\r\ntest app: https://github.com/appcelerator/alloy/tree/master/test/apps/testing/ALOY-884\r\n\r\nFunctional test\r\n\r\n\"Compile\" = \"alloy compile --platform ios\"\r\n\r\n# Compile the test app once. This should work without error.\r\n# Compile the test app again. This should again work without error.\r\n\r\nPrior to this fix, step 2 would cause the error shown in the description above. It is not necessary to do any runtime evaluation to confirm that this bug is fixed.",
					"updateAuthor": {
						"name": "tlukasavage",
						"key": "tlukasavage",
						"displayName": "Tony Lukasavage",
						"active": true,
						"timeZone": "America/Los_Angeles"
					},
					"created": "2013-11-23T14:56:27.000+0000",
					"updated": "2013-11-23T14:56:27.000+0000"
				},
				{
					"id": "280872",
					"author": {
						"name": "fokkezb",
						"key": "fokke",
						"displayName": "Fokke Zandbergen",
						"active": true,
						"timeZone": "Europe/Amsterdam"
					},
					"body": "It seemed fixed, but of course cleaning up is only relevant the second run and then it still crashes. When I run the ALOY-884 test using jake and then compile the generated Harness project a second or more time is passes. So I need to check what the difference is with my project(s).",
					"updateAuthor": {
						"name": "fokkezb",
						"key": "fokke",
						"displayName": "Fokke Zandbergen",
						"active": true,
						"timeZone": "Europe/Amsterdam"
					},
					"created": "2013-11-23T15:11:59.000+0000",
					"updated": "2013-11-23T15:27:08.000+0000"
				},
				{
					"id": "280873",
					"author": {
						"name": "fokkezb",
						"key": "fokke",
						"displayName": "Fokke Zandbergen",
						"active": true,
						"timeZone": "Europe/Amsterdam"
					},
					"body": "If you run just {{alloy compile --platform ios}} it passes, but when you run {{ti build -p ios}} multiple times it fails after the first build. This is true for both my project and the Harness project generated by {{jake app:run dir=testing/ALOY-884}}. Note that after generating the project you need to change to its root and run {{ti build -p ios}} from there because jake will just generate a new project every time.",
					"updateAuthor": {
						"name": "fokkezb",
						"key": "fokke",
						"displayName": "Fokke Zandbergen",
						"active": true,
						"timeZone": "Europe/Amsterdam"
					},
					"created": "2013-11-23T15:40:27.000+0000",
					"updated": "2013-11-23T15:40:27.000+0000"
				},
				{
					"id": "280876",
					"author": {
						"name": "tlukasavage",
						"key": "tlukasavage",
						"displayName": "Tony Lukasavage",
						"active": true,
						"timeZone": "America/Los_Angeles"
					},
					"body": "re-opening to look at [~fokke]'s case",
					"updateAuthor": {
						"name": "tlukasavage",
						"key": "tlukasavage",
						"displayName": "Tony Lukasavage",
						"active": true,
						"timeZone": "America/Los_Angeles"
					},
					"created": "2013-11-24T00:51:40.000+0000",
					"updated": "2013-11-24T00:51:40.000+0000"
				},
				{
					"id": "280898",
					"author": {
						"name": "tlukasavage",
						"key": "tlukasavage",
						"displayName": "Tony Lukasavage",
						"active": true,
						"timeZone": "America/Los_Angeles"
					},
					"updateAuthor": {
						"name": "tlukasavage",
						"key": "tlukasavage",
						"displayName": "Tony Lukasavage",
						"active": true,
						"timeZone": "America/Los_Angeles"
					},
					"created": "2013-11-24T15:39:35.000+0000",
					"updated": "2013-11-24T15:39:35.000+0000"
				},
				{
					"id": "281098",
					"author": {
						"name": "fokkezb",
						"key": "fokke",
						"displayName": "Fokke Zandbergen",
						"active": true,
						"timeZone": "Europe/Amsterdam"
					},
					"updateAuthor": {
						"name": "fokkezb",
						"key": "fokke",
						"displayName": "Fokke Zandbergen",
						"active": true,
						"timeZone": "Europe/Amsterdam"
					},
					"created": "2013-11-25T07:32:56.000+0000",
					"updated": "2013-11-25T07:33:32.000+0000"
				},
				{
					"id": "282690",
					"author": {
						"name": "fcasali",
						"key": "fcasali",
						"displayName": "Federico Casali",
						"active": true,
						"timeZone": "America/Los_Angeles"
					},
					"body": "Cannot reproduce also following latest comments from Tony and Fokke. \n\nTiSDK 3.2.0.v20131204220843\nCLI 3.2.0-beta\nAlloy 1.3.0-beta\n\nClosing.",
					"updateAuthor": {
						"name": "fcasali",
						"key": "fcasali",
						"displayName": "Federico Casali",
						"active": true,
						"timeZone": "America/Los_Angeles"
					},
					"created": "2013-12-05T20:26:58.000+0000",
					"updated": "2013-12-05T20:26:58.000+0000"
				}
			],
			"maxResults": 7,
			"total": 7,
			"startAt": 0
		}
	}
}